### DISCUSSION TITLE
Access violation error on logoff

### DISCUSSION THREAD

**Discussion by Anixx** [valinet/ExplorerPatcher#894] [General]
Now I experience absolutely persistent access violation message box (memory cannot be "read") in the explorer process at any logoff or shutdown of the system. This does not happen on the stable release.

**Comment by Anixx** [comment]
The access attempt is always to this address at the end of the address space:

![изображение](https://user-images.githubusercontent.com/1271736/154808211-76c354b8-9555-4bea-b556-41f43a1387a5.png)

**Reply by valinet** [reply]
Try at least to determine the last version that did not produce that.

**Reply by l0oo** [reply]
Hello
I have it too during shutdown/restart of the system... 
Dunno if both addresses are the same as on Anixx screenshot (pop-up is very short). But 0xFFFFFFFFFFF... is there for sure.

**Reply by valinet** [reply]
Do you use the classic theme as well? Try to debug it, I personally can't reproduce it.

**Reply by l0oo** [reply]
Yup. Classic theme here.
Will try to check which version was the latest good.

**Reply by valinet** [reply]
I also encountered that once on my classic theme VM, but couldn’t reproduce. Try to see if you can enable full crash dumps and whether it creates a crash dump for that error. If yes, then I can look on that, only make sure you’re on the latest pre-release so I know easier which symbols to use for that dump.

**Reply by OrthodoxWindows** [reply]
I have the same problem. I also use the classic theme. My system is Windows 10 20H2.
The Explorer Patcher version is **22000.493.42.16**.

**Reply by valinet** [reply]
I understand, but without any logs or additional pointers, I can't guess what may cause it, also considering that I can't even reproduce it. Anyway, I don't feel like it's such a big deal since it happens when the program terminates, but nevertheless, if we can diagnose it and solve it, the better.

**Reply by OrthodoxWindows** [reply]
I actually get an error when closing session, but it does not seem related to Explorer Patcher.
In fact, I had already this issue before the installation of Explorer Patcher.

I add files however : 

First Error:

``Niveau	Date et heure	Source	ID de l’événement	Catégorie de la tâche
Erreur	21/02/2022 17:52:30	Application Error	1000	Événements d’arrêts inattendus d’applications	"Nom de l’application défaillante LogonUI.exe, version : 10.0.19041.1, horodatage : 0xc08a5452
Nom du module défaillant : verifier.dll, version : 10.0.19041.1, horodatage : 0x3cbe5ecd
Code d’exception : 0xc0000421
Décalage d’erreur : 0x000000000000727a
ID du processus défaillant : 0xd6c
Heure de début de l’application défaillante : 0x01d8274369fc1847
Chemin d’accès de l’application défaillante : C:\WINDOWS\system32\LogonUI.exe
Chemin d’accès du module défaillant: C:\WINDOWS\System32\verifier.dll
ID de rapport : 23549294-4710-4b3a-8fc5-faec8cc5c831
Nom complet du package défaillant : 
ID de l’application relative au package défaillant : "

Second Error :

Niveau	Date et heure	Source	ID de l’événement	Catégorie de la tâche
Erreur	21/02/2022 17:52:30	Application Error	1000	Événements d’arrêts inattendus d’applications	"Nom de l’application défaillante LogonUI.exe, version : 10.0.19041.1, horodatage : 0xc08a5452
Nom du module défaillant : verifier.dll, version : 10.0.19041.1, horodatage : 0x3cbe5ecd
Code d’exception : 0xc0000421
Décalage d’erreur : 0x000000000000727a
ID du processus défaillant : 0xd6c
Heure de début de l’application défaillante : 0x01d8274369fc1847
Chemin d’accès de l’application défaillante : C:\WINDOWS\system32\LogonUI.exe
Chemin d’accès du module défaillant: C:\WINDOWS\System32\verifier.dll
ID de rapport : d768fc52-9be8-4be1-b4c1-e4a144cea6cc
Nom complet du package défaillant : 
ID de l’application relative au package défaillant : "

**Comment by Bec-de-Xorbin** [comment]
I can also reproduce this on two laptops, default theme, no classic. Definitely not a fix but if you don't like to see that popup, you can use the following registry change (save as .reg or change manually):
```
Windows Registry Editor Version 5.00

[HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\Windows]
"ErrorMode"=dword:00000002
```
Reference: https://docs.microsoft.com/en-us/troubleshoot/windows/win32/hard-error-popup-mode

**Reply by valinet** [reply]
It’s nice to know, but, I say it again, I’d more interested in a way to fix this.

**Comment by jed84** [comment]
This occurs when 7+ Taskbar Tweaker 5.12.3.4 is also installed.
Does not occur if 7+ Taskbar Tweaker 5.12.3.4 does not work in the background.

**Reply by valinet** [reply]
Okay, interesting. Will look into it, thanks.

**Comment by jed84** [comment]
I also verified this does *not* occur when the last stable ExplorerPatcher (22000.469.41.16) and Taskbar Tweaker 5.12.3.4 are installed.

Only the latest pre-releases of ExplorerPatcher (including 22000.493.42.17) are affected and on my machine 7+ Taskbar Tweaker 5.12.3.4 must work in the background to trigger the error message.

**Reply by valinet** [reply]
It has nothing to do with the classic theme.

**Reply by l0oo** [reply]
That is actually good, it is not coz of Classic Theme... :)
I can confirm. No 7+TT = no error.
Wonder if it is coincidence with some 7+TT/EP settings or just 7+TT in background is enough.

**Comment by OrthodoxWindows** [comment]
I confirm, the issue only happens with 7+ taskbar tweaker enabled. The 7+ version is 5.12.3. The EP version is 22000.493.42.26.

**Comment by OrthodoxWindows** [comment]
I get the issue again from the version 22000.493.42.30. Retrograde to the version 22000.493.42.29 solve the issue. All versions higher than 22000.493.42.30 are concerned.

![A0016](https://user-images.githubusercontent.com/100134023/158076664-2d1f2ad1-94ac-4124-9c2a-5460fa099345.png)

**Reply by valinet** [reply]
Do you use any of the following options and 7+TT?

![image](https://user-images.githubusercontent.com/6503598/158076847-8b8c7720-addb-4091-97f7-4d39178e0125.png)

**Reply by OrthodoxWindows** [reply]
Yes, i use the "pinned items act as quick lanch" option (i did not realize). Of course, I also use 7+ taskbar tweaker.
If I disable the option, the error message is always present, but closes automatically. This is not a perfect solution, but it suits me.
Thank you.

**Reply by valinet** [reply]
So the message always appears, regardless of the option? It closes automatically after a bit of time anyway...

**Reply by OrthodoxWindows** [reply]
On the last pre-release (22000.556.42.35), the issue is fixed. I was not on the last.
It's now perfect. Many Thank.
